src/common/models/Projection.ts

```diff
@@ -470,6 +477,22 @@ export class Projection {
     return _.mean(lastWeekOfPositives);
   }
 
+  getWeeklyAverageDeathForDay(i?: number) {
+    const lastIndex = this.indexOfLastValue(this.dailyDeaths);
+    if (i === undefined) {
+      if (!lastIndex) return null;
+      i = lastIndex;
+    } else if (lastIndex === null || i > lastIndex) {
+      return null;
+    }
+    // Get last week of sane data (ignore negative values)
+    const lastWeekOfDeaths = _.filter(
+      this.dailyDeaths.slice(Math.max(0, i - 6), i + 1),
+      p => p !== null && p >= 0,
+    );
+    return _.mean(lastWeekOfDeaths);
+  }
```

Comment:
I think you could combine getWeeklyAverageCaseForDay() and getWeeklyAverageDeathForDay() into a single `getWeeklyAverageForDay(data: Array<number|null>, i?: number)` helper.

And then the calling code would do `getWeeklyAverageForDay(this.dailyPositiveTests, ...)` or `getWeeklyAverageForDay(this.dailyDeaths, ...)` as appropriate.

FWIW- We _also_ have a `smoothWithRollingAverage()` helper that generates a new array with the averaged values.  I think we that would be more efficient and cleaner, but that's a bit of extra churn / effort (need to make sure it handles negative outliers in the same way), so that's probably best for a later project.

Reply:
Keeping them separate for now since getWeeklyAverageCaseForDay() is used in contact_tracing.ts. Adding a TODO to combine them